core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/uri/ComponentURI.java

```diff
@@ -99,6 +99,11 @@ public final boolean isUnspecified() {
 		return TerminologyRegistry.UNSPECIFIED.equals(resourceUri().getResourceId());
 	}
 	
+	@JsonIgnore
+	public final ResourceURIWithQuery asResourceURIWithQuery() {
+		return resourceUri().withQuery("ecl=" + identifier());
```

Comment:
We also have a `withQueryPart` that re-assembles the ResourceURI side from its building blocks, which one should be used and when?

https://github.com/b2ihealthcare/snow-owl/blob/98b927ea6ce2fe160e78edf0065f3d539f6d73b2/core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/ResourceURI.java#L188-L191

https://github.com/b2ihealthcare/snow-owl/blob/98b927ea6ce2fe160e78edf0065f3d539f6d73b2/core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/ResourceURI.java#L295-L297

Reply:
Yeah, I saw that, will rework this part later.